### DISCUSSION TITLE
STM32G0: USART flash and Serial Monitor via the same USART Interface or "How to use pin remap"

### DISCUSSION THREAD

**Discussion by olikraus** [stm32duino/Arduino_Core_STM32#1648] [Show and tell]
I am very much impressed by STM32duino.

Just like a normal AVR UNO Board I can flash and communicate with the board via serial interface. Only a small little trick is required (see below). I didn't found that trick in the Wiki for this project, so  this is why I created this discussion.

![IMG_20220203_201644104](https://user-images.githubusercontent.com/1084992/152642446-482369a2-7182-4c44-80b4-ec77b86af5cf.jpg)

USB-USART converter is connected to PA9 (TX) and PA10 (RX).

Flashing works as expected (issue https://github.com/stm32duino/Arduino_Core_STM32/issues/1637 needs to be considered). But how does the serial monitor work? I figured out, that we need to assign and remap the pins with setRx and setTx command.
The important trick is this: Pin names need an appended "_R" because those pins need to be remapped first.

``` C++
#define MYPIN 6

void setup() {
  Serial.setRx(PA_10_R);
  Serial.setTx(PA_9_R);
  Serial.begin(9600);
  pinMode(MYPIN, OUTPUT);
  Serial.println("setup done");
}

void loop() {
  Serial.println(millis());
  digitalWrite(MYPIN, HIGH);   // turn the LED on (HIGH is the voltage level)
  delay(300);                       // wait for a second
  digitalWrite(MYPIN, LOW);    // turn the LED off by making the voltage LOW
  delay(300);                       // wait for a second
}
```

Note that package pin 7 (PA_13) is mapped to Arduino digital pin 6.

**Comment by fpistm** [comment]
https://github.com/stm32duino/wiki/wiki/API#hardwareserial

**Reply by olikraus** [reply]
yes, but the mentoined wiki does not tell, that I have to remap PA_10 and PA_9, thats why I created this "show and tell" topic.

**Comment by fpistm** [comment]
So title should be: how to use remap pins of the STM32G0 😉

**Reply by olikraus** [reply]
Updated title :-)
